attrs_strict/_type_validation.py

```diff
@@ -68,6 +83,25 @@ def _validate_elements(attribute, value, expected_type):
         _handle_tuple(attribute, value, expected_type)
     elif base_type == typing.Union:
         _handle_union(attribute, value, expected_type)
+    elif base_type in {typing.Callable, Callable}:
+        _handle_callable(attribute, value, expected_type)
+
+
+def _handle_callable(attribute, callable_, expected_type):
+    _signature = signature(callable_)
+    callable_args = [
```

Comment:
Good question, the tests do work but that's because I am specifying `__annotations__`. 

For [pep484-Suggested syntax for Python 2.7](https://www.python.org/dev/peps/pep-0484/#suggested-syntax-for-python-2-7-and-straddling-code), it would not. I'll look into how we could support that. 🤔

Reply:
It seems like the comment-based-syntax for type checking `python2` code is not mandated by [pep484](https://www.python.org/dev/peps/pep-0484/#suggested-syntax-for-python-2-7-and-straddling-code).
>Some tools may want to support type annotations in code that must be compatible with Python 2.7. For this purpose this PEP has a suggested (but not mandatory) extension where function annotations are placed in a # type: comment

If we want to support that we would need to implement the parsing of it (I couldn't find a library that does that :/).  Alternatively, we could only support `__annotations__` based approach for `python2` which I think would be more preferred by a majority of users since it is `python2`/`python3` compatible. However, they'd still need to have the type comment if they want to use `mypy` with `python2`.



Please advise how you'd like to proceed?

Reply:
I think we should *not* support that for python 2 and mention it in the documentation. I think having it feature proof for python3 is what we should aim for so let's forget the idea of supporting type annotations for python2 wrt callables.